tools/haskell-language-server/default.nix

```diff
@@ -30,11 +30,25 @@ let
   fromSource = name: modules:
     let
       planConfig = planConfigFor name modules // {
-        src = sources."${name}";
+        src = sources'."${name}";
       };
     in
       allExes (haskell-nix.cabalProject planConfig)."${name}";
 
+  sources' = {
+    # This has a submodule, which niv doesn't yet handle.
+    # Note that this also defeats nix-prefetch-git.
+    # https://github.com/nmattia/niv/issues/229
+    haskell-language-server =
+      with sources.haskell-language-server;
+      fetchFromGitHub {
+        inherit owner repo rev;
+        name = "haskell-language-server-with-submodules-src";
+        sha256 = "0b94l6bywa6jk20y2cswyq5ks4g515895k2apvr1mdfkfhngdb7b";
```

Comment:
Instead of inheriting the hash that the tooling gives us, provide the right one here.  Quietly overriding it using niv the natural way contributed to the mess where it works for me locally but fails in CI.  Niv updates will fail until the linked enhancement is added to niv.

Also changed the name to emphasize that it's different.